dbt/adapters/spark/connections.py

```diff
@@ -22,9 +25,19 @@
 NUMBERS = DECIMALS + (int, float)
 
 
+def _build_odbc_connnection_string(**kwargs) -> str:
+    return ";".join([f"{k}={v}" for k, v in kwargs.items()])
+
+
 class SparkConnectionMethod(StrEnum):
     THRIFT = 'thrift'
     HTTP = 'http'
+    ODBC = 'odbc'
+
+
+class SparkClusterType(StrEnum):
+    ALL_PURPOSE = "all-purpose"
+    VIRTUAL = "virtual"
```

Comment:
Databricks isn't using the name "virtual clusters" anymore, I believe they're just calling them "endpoints."

Rather than a combination of `cluster` and `cluster_type`, I think we should make the distinction between `cluster` (old style, all-purpose/interactive) and `endpoint` (new style).